src/main/java/meditracker/dailymedication/DailyMedicationManager.java

```diff
@@ -71,34 +61,20 @@ public static void importDailyMedicationManager(List<String> lines) {
      * @param line each line read from the textfile
      */
     private static void parseImportedLine(String line) {
-        String[] fields = line.split("\\|");
-        boolean isTaken = Boolean.parseBoolean(fields[1].toLowerCase().trim());
-        DailyMedication dailyMedication = new DailyMedication(fields[2].trim());
-        if (isTaken) {
-            dailyMedication.take();
-        } else {
-            dailyMedication.untake();
+        DailyMedication dailyMedication = DailyMedication.fromStringData(line);
+        if (dailyMedication == null) {
+            return;
         }
-        addImportToSubLists(fields[0].toUpperCase().trim(), dailyMedication);
+        addImportToSubLists(dailyMedication);
     }
 
     /**
      * Imports data from the read text file
      *
-     * @param period time of the day
      * @param dailyMedication daily medication to be taken for the day to add to respective sub lists
      */
-    private static void addImportToSubLists(String period, DailyMedication dailyMedication) {
-        if (period.equals("M")) {
-            addDailyMedication(dailyMedication, Period.MORNING);
-        } else if (period.equals("A")) {
-            addDailyMedication(dailyMedication, Period.AFTERNOON);
-        } else if (period.equals("E")) {
-            addDailyMedication(dailyMedication, Period.EVENING);
-        } else {
-            MEDILOGGER.warning("Assigned medication period not recognised. \"" +
-                    dailyMedication.getName() + "\" not imported into today's list.");
-        }
+    private static void addImportToSubLists(DailyMedication dailyMedication) {
```

Comment:
This method appears to just be a wrapper for `addDailyMedication`. Might be better to just replace all usage of this method with `addDailyMedication` instead.

Reply:
Implemented in 1fb50b4